File: src/dataset/cpp/sieve_exercism-chrisnp/openAI/ChatGPT4_sieve_v4.cpp
#include "sieve.h"
#include <vector>

std::vector<int> sieve::primes(int limit) {
    if (limit < 2) return {};

    std::vector<bool> sieve(limit + 1, true);
    sieve[0] = sieve[1] = false;
    std::vector<int> primes;
    primes.reserve(limit / 2); // Reserve approximate space to reduce reallocations

    for (int candidate = 2; candidate * candidate <= limit; ++candidate) {
        if (sieve[candidate]) {
            for (int m = candidate * candidate; m <= limit; m += candidate) {
                sieve[m] = false;
            }
        }
    }

    for (int candidate = 2; candidate <= limit; ++candidate) {
        if (sieve[candidate]) {
            primes.push_back(candidate);
        }
    }

    return primes;
}

File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/openAI/ChatGPT4_sieve_v4.js
var Sieve = function(size) {
	this.primes = [];
	if (size < 2) return;

	const sieve = new Uint8Array(size + 1);
	sieve[2] = 1;
	this.primes.push(2);

	for (let n = 3; n <= size; n += 2) {
		if (!sieve[n]) {
			this.primes.push(n);
			for (let m = n * n; m <= size; m += n * 2) {
				sieve[m] = 1;
			}
		}
	}
};

module.exports = Sieve;